src/main/java/de/cronn/validation_files_diff/JoiningStrategy.java

```diff
@@ -35,11 +38,7 @@ public String getPrefix(PsiElement element) {
 	}
 
 	public Path addToPath(Path current, PsiElement element) {
-		Path resultingPath = current;
-		for (String part : getPrefix(element).split("/")) {
-			resultingPath = resultingPath.resolve(part);
-		}
-		return resultingPath;
+		return current.resolve(getPrefix(element));
```

Comment:
I prefer the solution with splitting on "/" as we then don't need to worry about file delimiters which may be system dependent. For example on Windows usually "\" is used instead - I am not sure if your proposal would handle this correctly.